src/identity/configuration/IdentityProviderFactory.ts

```diff
@@ -136,9 +139,44 @@ export class IdentityProviderFactory implements ProviderFactory {
     const provider = new Provider(this.baseUrl, config);
     provider.proxy = true;
 
+    this.alwaysRenderErrors(provider);
+
     return provider;
   }
 
+  /**
+   * In the `configureErrors` function below, we configure the `renderError` function of the provider configuration.
+   * This function is called by the OIDC provider library to render errors,
+   * but only does this if the accept header is HTML.
+   * Otherwise, it just returns the error object iself as a JSON object.
+   * See https://github.com/panva/node-oidc-provider/blob/0fcc112e0a95b3b2dae4eba6da812253277567c9/lib/shared/error_handler.js#L48-L52.
+   *
+   * In this function we override the `ctx.accepts` function
+   * to make the above code think HTML is always requested there.
+   * This way we have full control over error representation as configured in `configureErrors`.
+   * We still check the accept headers ourselves so there still is content negotiation on the output,
+   * the client will not simply always receive HTML.
+   *
+   * Should this part of the OIDC library code ever change, our function will break,
+   * at which point behaviour will simply revert to what it was before.
+   */
+  private alwaysRenderErrors(provider: Provider): void {
+    provider.use(async(ctx, next): Promise<void> => {
+      const accepts = ctx.accepts.bind(ctx);
+
+      // Using `any` typings to make sure we support all different versions of `ctx.accepts`
+      ctx.accepts = (...types: any[]): any => {
+        // Make sure we only override our specific case
```

Comment:
Why don't we override in general? If we always choose `html`, we will just always render it ourselves?

If not, then can we add a comment about which cases we are not overriding?

Reply:
We are breaking the `accepts` function here, returning `html` in this case is incorrect and is just done to fool the OIDC library to then call the `renderError` function. But the `accepts` function might also be used in other cases not tied to the `renderError` call. It's just a function that returns a string.

E.g., here it is is used to make sure the introspection call has the correct accept header or throws an error otherwise: https://github.com/panva/node-oidc-provider/blob/9da61e9c9dc6152cd1140d42ea06abe1d812c286/lib/actions/introspection.js#L71-L74

Returning `html` in that case would break the OIDC library.

Reply:
Ok. So we are really monkey patching. That's okay.

Reply:
Or alternatively `if (types.includes('html'))` (just to be more robust).